torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -59,16 +59,24 @@ static std::vector<std::string> TORCH_NCCL_DESYNC_DEBUG = {
     "TORCH_NCCL_DESYNC_DEBUG",
     "NCCL_DESYNC_DEBUG"};
 
+// Environment variable to enable whether to record and check the start
+// time of one collective on CUDA.
 static std::vector<std::string> TORCH_NCCL_ENABLE_TIMING = {
     "TORCH_NCCL_ENABLE_TIMING",
     "NCCL_ENABLE_TIMING"};
 
+// Environment variable to enable monitoring thread which abort the process
+// when no heartbeat is detected after TORCH_NCCL_HEARTBEAT_TIMEOUT_SEC.
```

Comment:
"Enable monitoring thread which aborts the process when the ProcessGroupNCCL Watchdog thread gets stuck, which can happen due to calling CUDA/NCCL APIs that may hang.  Useful to prevent jobs being stuck  for a longer time than necessary tying up cluster resources."